tests/unittests/test_importer.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+
+from cloudinit.importer import match_case_insensitive_module_name
+
+
+@pytest.mark.parametrize(
+    "m_name,m_match",
+    (
+        pytest.param(
+            "nocloud-net",
+            "DataSourceNoCloud",
+            id="nocloud-net is a special case, make sure it works",
```

Comment:
Not something that needs to change here, but beyond explaining the test case, these id's are convenient for specifying a particular test when running on the cli.

`pytest "tests/unittests/test_importer.py::test_importer[nocloud-net is a special case, make sure it works]"` works but could probably be more concise.